### ISSUE TITLE
Potential perf improvements

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/OrchardCore#16676]
While looking at display events I noticed some potential improvements

- `PlacementInfo` and its results (tab/group/alternates) could be reused since they don't change when created from a string source
- `ShapeDescriptor` uses collections that might be empty. Check if we can lazy-load them. Note that they should be already reused across tenants so maybe it's not much gain at least for the cost of instantiation, not memory usage which would still be gained.
- Alternate collections like the ones for content parts in `ContentPartDisplayDriver<TPart>` should remain constant across usages. Maybe these could be cached per part. Or at least used as an immutable collection such that other parts of the code can be based on it. Another option is to have a list of collections so one can be reused while others can be added dynamically.

**Comment by sebastienros** [comment]
Abusing this issue to share findings. Running an agent with the tools required to do benchmarking (dedicated servers) and getting traces.

First round:

> Done—using 3 baseline runs plus trace and 3 isolation experiments, the main bottleneck is rendering/shapes/templates (11,142 baseline RPS → 43,475 when returning static after content load, +290%), the
  secondary bottleneck is content retrieval/materialization (43,475 → 84,328 when skipping content load, +94%), and authorization is minor in this scenario (11,142 → 11,941, +7%).
  Recommended improvements: (1) reduce shape/template work on hot detail pages (fewer shape renders/alternates, flatten template pipeline, cache expensive render fragments), (2) reduce content-load overhead
  (cache hydrated article payload or a precomputed view model for about), and (3) target allocation reductions in BuildDisplayAsync/Liquid path (reuse objects, trim transient dictionaries/async state in hot
  loops).

**Comment by gvkries** [comment]
> * Alternate collections like the ones for content parts in `ContentPartDisplayDriver<TPart>` should remain constant across usages. Maybe these could be cached per part. Or at least used as an immutable collection such that other parts of the code can be based on it. Another option is to have a list of collections so one can be reused while others can be added dynamically.

I don’t think this is feasible. `Alternates` (and `Wrappers`) are created per shape and are intentionally mutable, since modules can add or remove entries at any point. Because the alternates depend on the specific rendering context (not just the part type) we need fresh collections for each shape.
In theory we could pool the lists, but we’d still need to clear and repopulate them every time, which likely results in similar overhead and adds complexity for little gain.
